src/utils/fetchWithTimeout.ts

```diff
@@ -21,9 +21,12 @@ function makeTimeout(timeoutMs: number) {
   };
 }
 
+/**
+ * @private
+ */
 export function fetchWithTimeout(
   url: string,
-  options: RequestOptions,
+  options: RequestInit,
```

Comment:
By dropping our own RequestOptions, which is a limited version of Fetch' RequestInit, we can not use fetchWithTimeout for more than whats defined on our own RequestOptions. Even though that might not cause issues, it felt like it was unneccesary.

By dropping this, this method does one thing only: `wrap a timeout around fetch`, and no longer limits the options to provide to fetch.